pkg/armhelpers/util.go

```diff
@@ -68,10 +68,10 @@ func K8sLinuxVMNameParts(vmName string) (poolIdentifier, nameSuffix string, agen
 	return vmNameParts[k8sLinuxVMAgentPoolNameIndex], vmNameParts[k8sLinuxVMAgentClusterIDIndex], vmNum, nil
 }
 
-// WindowsVMNameParts returns parts of Windows VM name e.g: 50621acs9000
+// WindowsVMNameParts returns parts of Windows VM name e.g: 50621k8s9000
 func WindowsVMNameParts(vmName string) (poolPrefix string, acsStr string, poolIndex int, agentIndex int, err error) {
-	poolPrefix = strings.Split(vmName, "acs")[0]
-	poolInfo := strings.Split(vmName, "acs")[1]
+	poolPrefix = strings.Split(vmName, "k8s")[0]
```

Comment:
We can't hardcode this anymore. Either we need to accept the value to split on or we need to accept acs, k8s, or aks.  Honestly to keep existing clusters working I think we have to support handling at least acs/k8s.

Reply:
added a regular expression